### DISCUSSION TITLE
Is there some way to auto-assign a random background color to each post?

### DISCUSSION THREAD

**Discussion by mozonte** [datenstrom/community#448] [Ask a question] [answered]
For a blog, I would like each post to have a unique background color (all pale colors so black text still is readable). Is there some way to make this happen at the blog-creation stage? (I'd like the colors to be permanently assigned, not generated randomly each time the site is loaded, so if you remember a post was blue, you can find that blue post again with your eyes)?

**Comment by schulle4u** [comment, marked as answer]
You could create different themes and assign the desired theme to each post. 

```
---
Title: Blue Blog Example
Published: 2019-09-03
Author: Example Name
Layout: blog
Theme: blue
---
```

See also: https://extensions.datenstrom.se/help/css-files

HTH
Steffen

**Comment by yomli** [comment]
As a hack, you could also apply a random color based on a hash of the title. To do so:

* edit `/system/layouts/blog.html`, line 2 as follow:
```html
<div class="content" style="background-color:#<?php echo substr(crc32($this->yellow->page->getHtml("titleContent")), 3, 4) ?>;  color: rgba(0,0,0,0.8)">
```
![](https://justpaste.it/img/small/2564e5220ffb7530da600dcd07440cdd.png)

* edit `/system/layouts/blogpages.html`, line 13 as follow:
```html
<div class="<?php echo $page->getHtml("entryClass") ?>" style="background-color: #<?php echo substr(crc32($page->getHtml("title")), 3, 4) ?>; color: rgba(0,0,0,0.8)">
```
![](https://justpaste.it/img/small/8331ab53f724adcc4b3d7c68e5e0d57e.png)

Of course, this is just an example. I recommend you make your own `blog.html` and `blogpages.html` layouts and your own CSS theme to get that right.

---
If you don't like the colors, use the crc32 hash to create a class:

* edit `/system/layouts/blog.html`, line 2 as follow:
```html
<div class="content bgColor-<?php echo substr(hash("crc32b",$this->yellow->page->getHtml("title")), -1) ?>">
```

* edit `/system/layouts/blogpages.html`, line 13 as follow:
```html
<div class="<?php echo $page->getHtml("entryClass") ?> bgColor-<?php echo substr(hash("crc32b", $page->getHtml("title")), -1) ?>">
```
* edit the CSS to add your own color for each of the 16 different `bgColor-*` classes:
```css
.bgColor-0 { background: black }
.bgColor-1 { background: red }
...
.bgColor-e { background: blue }
.bgColor-f { background: darkblue }
```
---
Note that here I used `substr(crc32($page->getHtml("title")), 3, 4)` in the first example, and `substr(hash("crc32b", $page->getHtml("title")), -1)` in the second example. This is because, by default, `crc32()` will produce a decimal integer. With this `substr()` setting, the result will be a pale color.
